SonarQube.MSBuild.Tasks/MergeRulesets.cs

```diff
@@ -0,0 +1,95 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="MergeRulesets.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace SonarQube.MSBuild.Tasks
+{
+    /// <summary>
+    /// MSBuild task to merge rulesets by adding "Include" references to a
+    /// specified ruleset
+    /// </summary>
+    public class MergeRulesets : Task
+    {
+        private static readonly XName IncludeElementName = "Include";
+        private static readonly XName PathAttributeName = "Path";
+        private static readonly XName ActionAttributeName = "Action";
+        private const string DefaultActionValue = "Default";
+
+        #region Input properties
+
+        /// <summary>
+        /// The path to the main ruleset file i.e. the one that will include the others
+        /// </summary>
+        [Required]
+        public string PrimaryRulesetFilePath { get; set; }
+
+        /// <summary>
+        /// The full file paths of any rulesets to include. Can be empty, in which case the task is a no-op.
+        /// </summary>
+        public string[] IncludedRulesetFilePaths { get; set; }
+
+        #endregion Input properties
+
+        #region Overrides
+
+        public override bool Execute()
+        {
+            if (!File.Exists(this.PrimaryRulesetFilePath))
+            {
+                throw new FileNotFoundException(Resources.MergeRulesets_MissingPrimaryRuleset, this.PrimaryRulesetFilePath);
+            }
+            if (this.IncludedRulesetFilePaths == null || this.IncludedRulesetFilePaths.Length == 0)
```

Comment:
If there is a property that returns the count/length then I use that in preference to using an extension method that uses an iterator.